src/mpol/fourier.py

```diff
@@ -218,56 +243,63 @@ class NuFFT(nn.Module):
 
     def __init__(
         self,
-        coords=None,
-        nchan=1,
-        uu=None,
-        vv=None,
-        sparse_matrices=True,
+        coords: GridCoords,
+        uu: NDArray[floating[Any]],
+        vv: NDArray[floating[Any]],
+        nchan: int = 1,
+        sparse_matrices: bool = True,
     ):
         super().__init__()
 
+        if not (same_uv := uu.ndim == 1 and vv.ndim == 1):
+            import warnings
```

Comment:
It seems like this will raise a warning if the user specifies `sparse_matrices=False`?

Reply:
Thank you for catching that! Looks like I did miss checking for sparsity there. I'll add that in.